File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/order/createOrderRequest.ts
import { DeliveryMethod, mapOrderBookToOrderBookRequest, Order, OrderBookRequest, PaymentMethod } from "../../../..";

export interface CreateOrderRequest {
    contactClientName: string,
    contactPhone: string,
    deliveryAddress: string,
    deliveryTime: Date,
    paymentMethod: PaymentMethod,
    deliveryMethod: DeliveryMethod,
    orderBooks: OrderBookRequest[]
}

export function mapOrderToCreateOrderRequest(order: Order): CreateOrderRequest {
    return {
        contactClientName: order.contactClientName,
        contactPhone: order.contactPhone,
        deliveryAddress: order.deliveryAddress,
        deliveryTime: order.deliveryTime,
        paymentMethod: order.paymentMethod,
        deliveryMethod: order.deliveryMethod,
        orderBooks: order.orderBooks.map(x => mapOrderBookToOrderBookRequest(x))
    }
}

export function getDefaultCreateOrderRequest(): CreateOrderRequest {
    return {
        contactClientName: "",
        contactPhone: "",
        deliveryAddress: "",
        deliveryTime: new Date(),
        paymentMethod: PaymentMethod.Cash,
        deliveryMethod: DeliveryMethod.AddressDelivery,
        orderBooks: []
    }
}

File: src/ELibrary.Backend/ShopApi/Features/OrderFeature/Dtos/CreateOrderRequest.cs
using LibraryShopEntities.Domain.Entities.Shop;

namespace ShopApi.Features.OrderFeature.Dtos
{
    public class CreateOrderRequest
    {
        public string ContactClientName { get; set; }
        public string ContactPhone { get; set; }
        public string DeliveryAddress { get; set; }
        public DateTime DeliveryTime { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public List<OrderBookRequest> OrderBooks { get; set; }
    }
}
